### PR TITLE
Add optional same_name_answer_ids key

### PR DESCRIPTION
**Pull request by iwootten**
### PR Context

Adds the same_name_answer_ids key in order to support tracking people with duplicate names in runner.

### CODE REVIEW

**Comment by iwootten** [comment]
[[T] Maintain duplicate first/last name list state (L)](https://trello.com/c/gbS9YV0C/4111-t-maintain-duplicate-first-last-name-list-state-l)

File: schemas/common_definitions.json [hunk 1]
```diff
@@ -302,6 +302,11 @@
     "type": "string",
     "minLength": 1
   },
+  "string_array": {
+    "type": "array",
+    "items": { "type": "string" },
+    "uniqueItems": true
```
Comment (MebinAbraham) [inline schemas/common_definitions.json:308]: Could add min items property.

**Review by MebinAbraham — DISMISSED** [review]

File: schemas/common_definitions.json (hunk 1 above)
Comment (MebinAbraham) [reply schemas/common_definitions.json:308]: Also string could reference "non" empty string definition.

**Comment by iwootten** [comment]
Going to update this to validate the same_same_answer_ids existence

File: schemas/common_definitions.json [hunk 2]
```diff
@@ -302,6 +302,14 @@
     "type": "string",
     "minLength": 1
   },
+  "string_array": {
+    "type": "array",
+    "items": {
+      "$ref": "https://eq.ons.gov.uk/common_definitions.json#/non_empty_string"
+    },
+    "uniqueItems": true,
+    "minLength": 1
```
Comment (MebinAbraham) [inline schemas/common_definitions.json:311]: Min length is no longer needed now that you validate the answer ids but `minItems` is needed as currently an empty array would validate successfully. 
https://json-schema.org/understanding-json-schema/reference/array.html#length

**Review by MebinAbraham — APPROVED** [review]

**Review by JamesGardiner — APPROVED** [review]

**Pull request merged**